src/Microsoft.DotNet.Build.Tasks/PackageFiles/clean.targets

```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <PropertyGroup>
+    <UserLocalFolder Condition="'$(OsEnvironment)'!='Unix'">$(LocalAppData)/</UserLocalFolder>
+    <UserLocalFolder Condition="'$(OsEnvironment)'=='Unix'">$(HOME)/.local/share/</UserLocalFolder>
+  </PropertyGroup>
+
+  <Target Name="CleanPackages">
+    <RemoveDir Directories="$(PackagesDir)" />
+  </Target>
+
+  <Target Name="CleanPackagesCache">
+    <RemoveDir Directories="$(UserLocalFolder)NuGet/Cache/;$(UserLocalFolder)NuGet/v3-cache/;$(UserLocalFolder)dnu/cache/" />
```

Comment:
You should test this but I suspect this will fail if one of those directories doesn't exist. You might need to do some exist checks before adding them.